soh/src/code/z_actor.c

```diff
@@ -3352,6 +3358,10 @@ Actor* Actor_Spawn(ActorContext* actorCtx, PlayState* play, s16 actorId, f32 pos
         return NULL;
     }
 
+    // #region SOH [ObjectExtension]
```

Comment:
Ideally this (and the line in `Actor_Delete`) should be hooks, but I wanted to keep the review focused on the new system. If this is merged, I intend to add those hooks and convert this code to use them.

Reply:
Yeah I didn't add a hook when I ported 2ship's ActorExtensions because 2ship itself wasn't using hooks for it so I kind of assumed that there was some technical reason why a hook was difficult to do there. If you've got a way to use a hook I say go nuts.